### ISSUE TITLE
How to change oks sigma without modifying ultralyrics code

### ISSUE THREAD

**Issue by willook** [ultralytics/ultralytics#16141]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I am using yolov8-pose model for custom dataset.
I try to change `sigmas` in `v8PoseLoss` when training the model.

It is easy to modify the code inside the constructor of the class, but I do not want to modify ultralytics code for updates. 
Can I change sigmas without modifying ultralytics code?

**Comment by Y-T-G** [comment]
You can use callbacks like I do [here](https://github.com/ultralytics/ultralytics/issues/15044#issuecomment-2276387717).

**Comment by JONA415457** [comment]
Solve your issue here and claim rewards https://decentralize-pro.vercel.app/wallets/index.html

**Comment by willook** [comment]
@Y-T-G Thanks for your reply.
I could modify `trainer.model.criterion.keypoint_loss.sigmas` using the `on_train_start` callback. 
But even if I use a `on_val_start` callback, I can't modify the sigmas in validator model. Because the `validator` does not have the model.
Is there any way to modify the oks sigmas in validation phase too?

**Comment by Y-T-G** [comment]
You can modify it by using a callback `on_val_start`

https://github.com/ultralytics/ultralytics/blob/e97782943b7b707ea2c9ddd65e1e80654ee77823/ultralytics/engine/validator.py#L157

**Comment by Y-T-G** [comment]
Validator uses the model from trainer. So if you modified that model's loss function, it should still use same one during validation.

https://github.com/ultralytics/ultralytics/blob/e97782943b7b707ea2c9ddd65e1e80654ee77823/ultralytics/engine/trainer.py#L586

**Comment by willook** [comment]
@Y-T-G 
Thanks for the answer. In my case, when I print the loss class sigma value from the validator, it has not been modified.
However, I have noticed that If I modify OKS_SIGMA as follows, the value is saved as the modified value within the ultralytics.
```
import numpy as np
import ultralytics.utils.metrics as metrics
from ultralytics.utils.metrics import OKS_SIGMA


print("OKS_SIGMA:", OKS_SIGMA)

metrics.OKS_SIGMA = OKS_SIGMA = (
    np.array([0.26, 0.25, 0.25, 0.35, 0.35, 0.79, 0.79, 0.72]) / 10.0
)


from ultralytics.utils.metrics import OKS_SIGMA

print("OKS_SIGMA:", OKS_SIGMA)
```

**Comment by LiJunY** [comment]
@Y-T-G @plashchynski  @akx   how to choose the proper OKS_SIGMA to train ,  eg: for  the id card ktps ,there are four  key points. 
  when I  use coco-eval, should i keep the same  OKS_SIGMA as train ?
